bench/hist/Main.hs

```diff
@@ -312,6 +332,18 @@ findPrev name (x : y : xx)
   | otherwise = findPrev name (y : xx)
 findPrev name _ = name
 
+data BuildSystem = Cabal | Stack
+  deriving (Eq, Read, Show)
+
+instance FromJSON BuildSystem where
+    parseJSON x = fromString . map toLower <$> parseJSON x
```

Comment:
I'd use Data.List.Extra.lower instead of map toLower, but your call.

Reply:
Why?

Reply:
Two reasons. 1) shorter. 2) turning a string lowercase is fundamentally not the same as turning each character lowercase individually - some characters become two characters in lowercase. Currently lower is not correct, but one day we can fix lower, while map toLower is never going to be right. In practice, for "cabal" and "stack" you are fine.